### PR TITLE
feat: Add APIs to add and revoke space-level access

### PR DESCRIPTION
**Pull request by yu-iskw**
Enhances: https://github.com/lightdash/lightdash/issues/5666

### Description:
Added methods.

- `POST /projects/{projectUuid}/spaces/{spaceUuid}/share`
- `DELETE /projects/{projectUuid}/spaces/{spacesUuid}/share/{userUuid}`

### What we have to do in the pull request
- [x] Update `reouts.ts` and `swagger.json`

### CODE REVIEW

File: packages/backend/src/controllers/spaceController.ts [hunk 1]
```diff
@@ -152,4 +152,64 @@ export class SpaceController extends Controller {
...
+     */
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('200', 'Success')
+    @Post('{spaceUuid}/access')
+    @OperationId('AddSpaceShareToUser')
+    @Tags('Roles & Permissions')
```
Comment (rephus) [inline packages/backend/src/controllers/spaceController.ts:171]: Nice tag

**Review by rephus — CHANGES_REQUESTED** [review]
Thanks for your contribution. 

The new endpoints look good, but we should replace the existing endpoint from `projectRouter`and use the same endpoints 

- add space share: https://github.com/lightdash/lightdash/blob/main/packages/backend/src/routers/projectRouter.ts#L403
- remove space share: https://github.com/lightdash/lightdash/blob/main/packages/backend/src/routers/projectRouter.ts#L420

File: packages/backend/src/controllers/spaceController.ts [hunk 2]
```diff
@@ -152,4 +153,64 @@ export class SpaceController extends Controller {
...
+     * @param userUuid The uuid of the user to grant access to
+     * @param req
+     */
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('200', 'Success')
+    @Post('{spaceUuid}/share')
```
Comment (yu-iskw) [inline packages/backend/src/controllers/spaceController.ts:170]: I have changed the path.

File: packages/common/src/types/space.ts [hunk 3]
```diff
@@ -43,3 +43,5 @@ export type ApiSpaceResponse = {
     status: 'ok';
     results: Space;
 };
+
+export type AddSpaceShare = Pick<SpaceShare, 'userUuid'>;
```
Comment (yu-iskw) [inline packages/common/src/types/space.ts:47]: I added a new type for the body of `addSpaceShare`.

**Review by yu-iskw — COMMENTED** [review]
@rephus Thank you for the comments. I have changed the paths in the controller so that those follows the router.

**Comment by rephus** [comment]
Great @yu-iskw we also need to remove the endpoints in the old "router" we don't want to have duplicated items. Once we do that, we can test this new endpoints on our preview environment to see if the behaviour is the same

**Comment by yu-iskw** [comment]
@rephus Thank you for merging! I look forward to releasing the APIs to Lightdash Cloud.

**Pull request merged**